docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+
+### React dependencies
+
+Run `npm install --save lodash chart.js`
+
+### CLI packaging command
+
+Run the `ayx_plugin_cli create-yxi` command, which bundles all the plugins in the workspace into a `.yxi` archive. It should look something like this:
+
+```powershell
+$ create-yxi
+[Creating YXI] started
+[Creating YXI] -- generate_config_files:generate_config_xml
+// output truncated //
+[Creating YXI] .  create_yxi:create_yxi
+[Creating YXI] finished
+```
+
+The resulting `.yxi` archive should exist in your plugin workspace's build/exit directory.
+
+## Install and Run in Designer
+
+### Method 1
+After you create a `.yxi,` you can double-click the `.yxi` or drag and drop it into Designer to install it. It then prompts you to install the package in a new dialog box. Which looks something like this:
+
+![YXI Install Dialog](./assets/install-yxi-dialog.jpg)
+
+Once it installs, you can find the plugin under the `Tensorflow Python SDK Examples` tool category.
+
+### Method 2
+You can also create the `.yxi` _**and**_ install it all in one step via the `designer-install` command. Choose the install option that matches your Designer install. Typically, this is the `user` install option. 
+
+
+
+Once the command finishes, open Designer and find your tool under the `Tensorflow Python SDK Examples` tool category.
+
+Drag it into your workflow. Using the recommended dataset (need a reference list, and include this [E-Note: <-]) at your chosen location, fill in the form similar to below:
+
+[TODO: SS of it. Have testing ones, in case]
+
+**TODO/ E-Note: add a selector to the UI; last second thought - debugging or toggling in the shiv directory sucks, and I don't want that smell in the guide—just a toggle switch on the UI for each mode. I forgot to circle back and add that earlier. Quick lift, like a one**
+
+(once it's in) Use the "MODE" selector to run them in the following order, for reasons described at the beginning of the guide: `DATA` -> `PREVIEW` -> `TRAIN` -> `PREDICT.`
+
+Outputs should resemble the following:
+
+#### Figure: `DATA` mode:
+
+[New SS]
+
+#### Figure: `PREVIEW` mode:
+
+[New SS]
+
+#### Figure: `TRAIN` mode:
+
+[New SS]
+
+#### Figure: `PREDICT` mode:
+
+[New SS]
+
+## Congratulations!
+
+You have successfully created your first Tensorflow Keras NN model, from scratch, in a workflow! If you're up to the challenge, we recommend the following exercises to test your new skills!
+
+### Exercises
+
+1. Create a second plugin for your production model.
+   1. Try locally using your production model by referencing it on your hard drive.
+   2. Convert the above to use a production model included in the yxi as a standalone deployable. (See the Pythons packaging module for idiomatic ways to include your model!)
+2. Add a FileExplorer to our text input fields for more intuitive data source input.
+3. Add an option to _continue_ training a model rather than overwriting each time.
+4. Convert the Tensorflow companion piece exercises.
```

Comment:
TW review:

1. Create a second plugin for your production model.
   1. Try using your production model locally by referencing it on your hard drive.
   2. Convert the above to use a production model included in the YXI as a standalone deployable. Refer to the Python packaging module for idiomatic ways to include your model.
2. Add a FileExplorer to the text input fields for more intuitive data source input.
3. Add an option to _continue_ training a model rather than overwriting each time.
4. Convert the Tensorflow companion piece exercises.